olp-cpp-sdk-dataservice-read/src/repositories/PartitionsCacheRepository.cpp

```diff
@@ -357,8 +357,17 @@ PartitionsCacheRepository::CreatePartitionKeys(
 }
 
 bool PartitionsCacheRepository::Protect(
-    const std::string& partition_id, const boost::optional<int64_t>& version) {
-  const auto keys = CreatePartitionKeys(partition_id, version);
+    const std::vector<std::string>& partition_ids,
+    const boost::optional<int64_t>& version) {
+  cache::KeyValueCache::KeyListType keys;
+  std::for_each(partition_ids.cbegin(), partition_ids.cend(),
+                [&](const std::string& partition_id) {
+                  auto partition_keys =
+                      CreatePartitionKeys(partition_id, version);
+                  std::move(partition_keys.begin(), partition_keys.end(),
+                            std::back_inserter(keys));
```

Comment:
I'm not sure that it is worth it but could we preallocate `keys` (assume successful `GetPartitionHandle` calls)? Should be `partition_ids.size() * 2u`.
Or at least verify that this approach is on par with the previous one in terms of speed.